### ISSUE TITLE
Build fails with vst-sdk 2.4

### ISSUE THREAD

**Issue by nick87720z** [muse-sequencer/muse#1014]
Muse 4.0.0 and git version can't build with official vst-sdk, but build with vestige.
Build log: [build.log](https://github.com/muse-sequencer/muse/files/6556353/build.log)

Steps to reproduce the behavior (in shell):
```sh
vst_header_path= # Your installation path for vst-sdk 2.4 (I don't expect 2.3 to be found anymore)
vst_header_path="${vst_header_path}/pluginterfaces/vst2.x"
cmake .. \
-DENABLE_{ALSA=off,DSSI=off,FLUID=off,INSTPATCH=off,LASH=off,LRDF=off,LV2=off,LV2_DEBUG=OFF,LV2_GTK2=off,OSC=off,PYTHON=off,RTAUDIO=off,RUBBERBAND=off,VST_{NATIVE=on,VESTIGE=off}} \
-DMODULES_BUILD_STATIC=off -DUPDATE_TRANSLATIONS=off -DCMAKE_BUILD_TYPE=Release \
-DVST_HEADER_PATH="${vst_header_path}"
make # Fail
```

I have vst-sdk-2.4 packaged in personal gentoo overlay, ebuild is [there](https://github.com/nick87720z/gentoo-overlay/blob/master/media-libs/vst-sdk/vst-sdk-2.4-r1.ebuild). Installed files layout repeats layout from dist archive. It installs to `/usr/include/vst24`, but since MusE expects `aeffectx.hpp` right in VST_HEADER_PATH, and this is the only file in vestige subdir, I set header path to `/usr/include/vst24/pluginterfaces/vst2.x`. For comparison, drumgizmo builds without problem, moreover - it expects right such layout (it takes header path `/usr/include/vst24`). I can't test it with sdk version 2.3, because unable to find anything downloadable (2.4 was in archive last time).

gentoo, gcc 7.4.0 (yup, I'm outdated, but DG doesn't complain yet)

**Comment by terminator356** [comment]
Try it now. Fixed (maybe) in git master.

1: Even if the error did not occur, compilation would never have made it past a couple of glaring errors,
 for example the StringLengths enumeration in plugin_cache_writer.h

2: There are a couple of __cdecl usages, for example in vst_native.h, that depend on a
 definition SDK_QUIRK which was not being exported from cmake texts to the code. (Fixed in config.h.in).

I have fixed the errors.

3: Actually, once I performed fix number 1, I did not require fix number 2. It worked fine without fix number 2.

So, maybe your results will be unchanged. You might have a slightly different SDK version.
There are some SDK deprecations in your build output that I don't have.
Try it...
